low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -169,22 +173,29 @@ def __init__(self, unit):
         self.previous_substate = "N/A"
         self.previous_pid = "N/A"
         self.nonactive_enter_timestamp = time.time()
+        self.active_enter_timestamp = time.time()
         self.nonactive_threshold = int(
             Conf.get(SSPL_CONF,
                      f"{ServiceMonitor.name().upper()}>threshold_inactive_time",
                      '60'))
+        self.active_threshold = int(Conf.get(SSPL_CONF,
+                     f"{ServiceMonitor.name().upper()}>threshold_active_time",
+                     '30'))
         self.properties_changed_signal = None
         self._service_state = ActiveState
         self._unit_state = None
 
     @property
     def is_enabled(self):
-        return 'disabled' not in self.properties_iface.Get(
+        return 'enabled' in self.properties_iface.Get(
```

Comment:
Modified condition here as - 
self.properties_iface.Get(UNIT_IFACE, 'UnitFileState') returns nothing if service UnitFile is not present for service (Services which are not configured )
For such services previous condition return TRUE and we are getting alerts, event if service is not present on setup.

Reply:
If a service, which needs to be monitored, is not configured, it will be not able recognized i believe. How/why this condition should return 'enabled' then in such case ?

Reply:
**'disabled' not in self.properties_iface.Get(UNIT_IFACE, 'UnitFileState')** 
Services which are present/configured on system only for those services this condition works.

for eg: 
statsd.service is configured and in enabled/disabled state, then this condition will work as expected.
But if 'lnet.service' is not configured on system/ (service unit is not present) in that case,
'properties_iface.Get(UNIT_IFACE, 'UnitFileState')' this return nothing, and 
'disbaled not in properties_iface.Get(UNIT_IFACE, 'UnitFileState') ' this condition will be true, 
and we get the alert for service even that is not configured on system.

```
>>> service_name=lnet.service
>>> str(properties_iface.Get(UNIT_IFACE, 'UnitFileState'))
''
>>> 'disabled' not in properties_iface.Get(UNIT_IFACE, 'UnitFileState')
True
>>> 'enabled' in properties_iface.Get(UNIT_IFACE, 'UnitFileState')
False
>>>
```

As per my understanding from monitoring_service list , services which are in enabled state only those services need to be monitored.
If any service from monitoring_service list is not configured on system, do we need to raise fault alert for such services??
if yes then in that case previous('disabled' not in self.properties_iface.) condition will work.

Reply:
[UnitFileState](https://www.freedesktop.org/wiki/Software/systemd/dbus/) can have these values: 
enabled, enabled-runtime, linked, linked-runtime, masked, masked-runtime, static, disabled

If except for disabled, a service will be classified as enabled for all other states. So,
```
'enabled' in self.properties_iface.Get(UNIT_IFACE, 'UnitFileState')
```
may not always give right results. 

In node health, I put additional check, if `command_line_path` exists for a service or not. If a service is not configured it wont exist.

```
path_array = properties_iface.Get(const.SERVICE_IFACE, 'ExecStart')
try:
    command_line_path = str(path_array[0][0])
except IndexError as err:
    logger.error(self.log.svc_log(f"Unable to find {service_name} path due to {err}"))
    command_line_path = "NA"
```

Reply:
Okay Thanks @jjxsg , I will add same comand_line_path check here to check if service is configured or not.
```
try:
            command_line_path = str(path_array[0][0])
        except IndexError as err:
            logger.error(
                "Unable to find %s path due to %s" % (self.name, err))
            command_line_path = "NA"
        if command_line_path != "NA":
            return 'disabled' not in self.properties_iface.Get(
                UNIT_IFACE, 'UnitFileState')
        else:
            return False
```

Reply:
Hi @shriya-deshmukh , @jjxsg  I am bit confused here, we need review from @sandeepsgt about this.

Reply:
@shriya-deshmukh  what if you add check like below

if self.properties_iface.Get(UNIT_IFACE, 'UnitFileState'):
    return 'disabled' not in self.properties_iface.Get(.....
else:
    return False

Reply:
Yes, It works. 
Added code change.